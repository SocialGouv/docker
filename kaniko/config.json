{
  "auths": {
    "harbor.fabrique.social.gouv.fr": {
      "username": "$HARBOR_USERNAME",
      "password": "$HARBOR_PASSWORD"
    }
  }
}
