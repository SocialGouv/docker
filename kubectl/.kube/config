apiVersion: v1
clusters:
- cluster:
    server: ${K8S_SERVER}
  name: ${K8S_CLUSTER_NAME}
contexts:
- context:
    cluster: ${K8S_CLUSTER_NAME}
    namespace: ${K8S_NAMESPACE}
    user: ${K8S_USERNAME}
  name: ${K8S_CLUSTER_NAME}
current-context: ${K8S_CONTEXT}
kind: Config
preferences: {}
users:
- name: ${K8S_USERNAME}
  user:
    token: ${K8S_TOKEN}