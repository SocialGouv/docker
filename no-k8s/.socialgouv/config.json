{
  "name": "myapp",
  "type": "app",
  "subdomain": "myapp"
}
