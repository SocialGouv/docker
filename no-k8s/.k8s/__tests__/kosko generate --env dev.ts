//

import { getEnvManifests } from "@socialgouv/kosko-charts/testing"
import { project } from "@socialgouv/kosko-charts/testing/fake/gitlab-ci.env"

import { name } from "../../.socialgouv/config.json"

jest.setTimeout(1000 * 60)
test("kosko generate --dev", async () => {
  expect(await getEnvManifests("dev", "", project(name).dev)).toMatchSnapshot()
})
