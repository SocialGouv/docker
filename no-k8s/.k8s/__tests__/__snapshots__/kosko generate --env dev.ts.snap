// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kosko generate --dev 1`] = `
"---
metadata:
  annotations:
    field.cattle.io/creatorId: gitlab
    field.cattle.io/projectId: ''
    git/branch: master
    git/remote: >-
      https://gitlab-ci-token:[MASKED]@gitlab.factory.social.gouv.fr/SocialGouv/myapp.git
    app.gitlab.com/app: socialgouv-myapp
    app.gitlab.com/env: master-dev2
    app.gitlab.com/env.name: master-dev2
  labels:
    azure-pg-admin-user: myapp
    application: master-dev2-myapp
    owner: myapp
    team: myapp
    cert: wildcard
  name: myapp-85-master-dev2
apiVersion: v1
kind: Namespace
---
metadata:
  annotations:
    app.gitlab.com/app: socialgouv-myapp
    app.gitlab.com/env: master-dev2
    app.gitlab.com/env.name: master-dev2
  labels:
    app: myapp
    application: master-dev2-myapp
    owner: myapp
    team: myapp
    cert: wildcard
  name: myapp
  namespace: myapp-85-master-dev2
spec:
  replicas: 1
  selector:
    matchLabels:
      app: myapp
  template:
    metadata:
      annotations:
        app.gitlab.com/app: socialgouv-myapp
        app.gitlab.com/env: master-dev2
        app.gitlab.com/env.name: master-dev2
      labels:
        app: myapp
        application: master-dev2-myapp
        owner: myapp
        team: myapp
        cert: wildcard
    spec:
      containers:
        - image: >-
            harbor.fabrique.social.gouv.fr/undefined/myapp:0123456789abcdefghijklmnopqrstuvwxyz0123
          livenessProbe:
            failureThreshold: 6
            httpGet:
              path: /healthz
              port: http
            initialDelaySeconds: 30
            periodSeconds: 5
            timeoutSeconds: 5
          name: myapp
          ports:
            - containerPort: 3000
              name: http
          readinessProbe:
            failureThreshold: 15
            httpGet:
              path: /healthz
              port: http
            initialDelaySeconds: 0
            periodSeconds: 5
            successThreshold: 1
            timeoutSeconds: 1
          resources:
            limits:
              cpu: 200m
              memory: 256Mi
            requests:
              cpu: 50m
              memory: 128Mi
          startupProbe:
            failureThreshold: 12
            httpGet:
              path: /healthz
              port: http
            periodSeconds: 5
apiVersion: apps/v1
kind: Deployment
---
metadata:
  labels:
    app: myapp
    application: master-dev2-myapp
    owner: myapp
    team: myapp
    cert: wildcard
  name: myapp
  annotations:
    app.gitlab.com/app: socialgouv-myapp
    app.gitlab.com/env: master-dev2
    app.gitlab.com/env.name: master-dev2
  namespace: myapp-85-master-dev2
spec:
  ports:
    - name: http
      port: 80
      targetPort: 3000
  selector:
    app: myapp
  type: ClusterIP
apiVersion: v1
kind: Service
---
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
    app.gitlab.com/app: socialgouv-myapp
    app.gitlab.com/env: master-dev2
    app.gitlab.com/env.name: master-dev2
  labels:
    app: myapp
    application: master-dev2-myapp
    owner: myapp
    team: myapp
    cert: wildcard
  name: myapp
  namespace: myapp-85-master-dev2
spec:
  rules:
    - host: master-dev2-myapp.dev2.fabrique.social.gouv.fr
      http:
        paths:
          - backend:
              serviceName: myapp
              servicePort: 80
            path: /
  tls:
    - hosts:
        - master-dev2-myapp.dev2.fabrique.social.gouv.fr
      secretName: wildcard-crt
apiVersion: extensions/v1beta1
kind: Ingress
"
`;
